src/MongoDB.Driver/ClientEncryption.cs

```diff
@@ -110,23 +113,24 @@ public BsonBinaryData Encrypt(BsonValue value, EncryptOptions encryptOptions, Ca
         /// <returns>The encrypted value.</returns>
         public async Task<BsonBinaryData> EncryptAsync(BsonValue value, EncryptOptions encryptOptions, CancellationToken cancellationToken)
         {
-            var bytes = value.ToBson(); //todo:
+            byte[] bytes = new BsonDocument("v", value).ToBson(serializer: BsonValueSerializer.Instance);
             return await _libMongoCryptController.EncryptFieldAsync(bytes, encryptOptions, cancellationToken).ConfigureAwait(false);
         }
 
-        private IKmsKeyId ParseKmsKeyId(BsonDocument masterKey)
+        private IKmsKeyId GetKmsProvider(string kmsProvider)
         {
-            if (!masterKey.TryGetValue("key", out var customerMasterKey))
+            switch (kmsProvider)
             {
-                throw new ArgumentException("TODO");
+                case "aws":
+                    throw new NotImplementedException();
```

Comment:
This is last-minute rollback from me. I didn't have time to investigate it for now. So, I temporarily commented out the `aws` processing. Not sure what object instance should provide data for parsing here. 

>DataKeyOptions dataKeyOptions 

from `CreateDataKey` method arguments, or options from here: https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/48/commits/ec0fce2ae90ae6c68bd1ac9995851d1963e10bdf#diff-4cfec9d614ac6a7f6769fe3627f58e76R34